core/consensus/aura/src/lib.rs

```diff
@@ -650,6 +654,27 @@ impl<B: Block, C, E> Verifier<B> for AuraVerifier<C, E> where
 	}
 }
 
+impl<B, C, E> Authorities<B> for AuraVerifier<C, E> where
+	B: Block,
+	C: ProvideRuntimeApi,
+	C::Api: AuthoritiesApi<B>,
+{
+	type Error = ConsensusError;
+
+	fn authorities(&self, at: &BlockId<B>) -> Result<Vec<AuthorityIdFor<B>>, Self::Error> {
+		authorities(self.client.as_ref(), at)
+	}
+}
+
+fn authorities<B, C>(client: &C, at: &BlockId<B>) -> Result<Vec<AuthorityIdFor<B>>, ConsensusError> where
+	B: Block,
+	C: ProvideRuntimeApi,
+	C::Api: AuthoritiesApi<B>,
+{
+	client.runtime_api().authorities(at)
```

Comment:
Could you please use the cache here (if miss => fallback to runtime call)? Is this possible to use it here?

Reply:
Done!
This was harder than I expected and brought even more changes. :sweat_smile: